source/client/process_impl.cc

```diff
@@ -246,7 +246,10 @@ void ProcessImpl::createBootstrapConfiguration(envoy::config::bootstrap::v2::Boo
     // We do not support any retrying.
     thresholds->mutable_max_retries()->set_value(0);
     thresholds->mutable_max_connections()->set_value(options_.connections());
-    thresholds->mutable_max_pending_requests()->set_value(options_.maxPendingRequests());
+    // We specialize on 0 below, as that is not supported natively. The benchmark client will track
+    // in flight work and avoid creating pending requests in this case.
+    thresholds->mutable_max_pending_requests()->set_value(
+        options_.maxPendingRequests() == 0 ? 1 : options_.maxPendingRequests());
```

Comment:
I didn't understand this. The circuit breaker doesn't allow a value of 0?

If the user passed --max_pending_requests 0 on the command line, meaning disable client-side queueing, then we set maxPendingRequests to 1 here. Does that mean we actually allow client-side queueing with a queue size of 1?

Reply:
Correct, the circuit breaker does not allow a value of 0. 

When 0 is specified for max-pending-requests in NH, we configure the circuit breaker to its minimum, 1. 
But: in the current mode (closed-loop) we actually attempt to avoid relying on the circuit breaker, because the idea is that we perform that gating ourselves so we can get a sense of  time spend being blocked on waiting for resources.
 
In the upcoming open-loop mode [1], Nighthawk won't try to avoid hitting the circuit breaker limits. A configured value of `0` should result in an error or warning there, because we'll solely rely on the circuit breaker.  

[1] https://github.com/envoyproxy/nighthawk/pull/177